File: 2015/solutions/day2.py
with open('../inputs/input2.txt', 'r') as f:
    data = [[int(val) for val in box.split('x')] for box in f.readlines()]

    # part one
    total = 0
    for box in data:
        products = [box[0] * box[1], box[0] * box[2], box[1] * box[2]]
        total += min(products)
        total += 2 * (sum(products))
    print(total)

    # part two
    ribbon = 0
    for box in data:
        ribbon += box[0] * box[1] * box[2]
        box.remove(max(box))
        ribbon += 2 * sum(box)
    print(ribbon)

File: 2017/solutions/day2.js
const fs = require('fs');

fs.readFile('../inputs/input2.txt', 'utf-8', (err, data) => {
    data = data.trim().split('\n').map(x => x.split('\t').map(y => parseInt(y)));
    let sum = data.map(x => Math.max(...x) - Math.min(...x)).reduce((a, b) => a + b);

    //part one
    console.log(sum);

    sum = data.map(x => {
        for (let i = 0; i < x.length - 1; i++) {
            for (let j = i + 1; j < x.length; j++) {
                if (x[i] % x[j] === 0) {
                    return x[i] / x[j];
                } else if (x[j] % x[i] === 0) {
                    return x[j] / x[i];
                }
            }
        }
    }).reduce((a, b) => a + b);

    //part two
    console.log(sum);
});